src/wallet/interfaces.cpp

```diff
@@ -550,7 +572,11 @@ class WalletImpl : public Wallet
     bool hdEnabled() override { return m_wallet->IsHDEnabled(); }
     bool canGetAddresses() override { return m_wallet->CanGetAddresses(); }
     bool privateKeysDisabled() override { return m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS); }
-    CoinJoin::Client& coinJoin() override { return m_coinjoin; }
+    CoinJoin::Client& coinJoin() override
+    {
+        if (m_coinjoin_client == nullptr) m_coinjoin_client = MakeCoinJoinClient(m_cjwalletman, *m_wallet);
+        return *m_coinjoin_client;
+    }
```

Comment:
Is this thread safe?

Reply:
no, but this code is droppped in 52a68427f721b8de4e5c23b170f63ebc584ce8db